File: output/CodeGen/codenet/C++/Java/s123678341.java
public class s123678341 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int X = in.nextInt();
        int Y = in.nextInt();
        int Z = in.nextInt();
        int K = in.nextInt();
        long[] A = new long[X];
        long[] B = new long[Y];
        long[] C = new long[Z];
        for (int i = 0; i < X; i++) {
            A[i] = in.nextLong();
        }
        for (int i = 0; i < Y; i++) {
            B[i] = in.nextLong();
        }
        for (int i = 0; i < Z; i++) {
            C[i] = in.nextLong();
        }
        Arrays.sort(A);
        Arrays.sort(B);
        Arrays.sort(C);
        long[] AB = new long[X * Y];
        for (int i = 0; i < X; i++) {
            for (int j = 0; j < Y; j++) {
                AB[i * Y + j] = A[i] + B[j];
            }
        }
        Arrays.sort(AB);
        long[] ABC = new long[min(K, AB.length)];
        for (int i = 0; i < min(K, AB.length); i++) {
            for (int j = 0; j < Z; j++) {
                ABC[i] += AB[i * Z + j] + C[j];
            }
        }
        Arrays.sort(ABC);
        for (int i = 0; i < K; i++) {
            System.out.println(ABC[i]);
        }
    }
}

File: output/CodeGen/codenet/C++/Python/s123678341.py
#!/usr/bin/python

import sys
import math
import random
import string
import time
import datetime
import os

def main():
    # read input
    X = int(sys.argv[1])
    Y = int(sys.argv[2])
    Z = int(sys.argv[3])
    K = int(sys.argv[4])
    
    A = []
    for i in range(X):
        A.append(int(sys.argv[5 + i]))
    B = []
    for i in range(Y):
        B.append(int(sys.argv[5 + X + i]))
    C = []
    for i in range(Z):
        C.append(int(sys.argv[5 + X + Y + i]))
    
    # sort
    A = sorted(A)
    B = sorted(B)
    C = sorted(C)
    
    # calculate
    AB = []
    for i in range(X):
        for j in range(Y):
            AB.append(A[i] + B[j])
    
    AB = sorted(AB)
    ABC = []
    for i in range(min(K, len(AB))):
        for j in range(Z):
            ABC.append(AB[i] + C[j])
    
    ABC = sorted(ABC)
    
    # output
    for i in range(K):
        print ABC[i]
    
if __name__ == '__main__':
    start = datetime.datetime.now()
    main()
    delta = datetime.datetime.now() - start
    print'real    time:', delta.seconds + 1e-6 * delta.microseconds